packages/salesforcedx-utils-vscode/src/commands/commandletExecutors.ts

```diff
@@ -159,9 +167,12 @@ export abstract class LibraryCommandletExecutor<T>
       const success = await vscode.window.withProgress(
         {
           title: nls.localize('progress_notification_text', this.executionName),
-          location: vscode.ProgressLocation.Notification
+          location: vscode.ProgressLocation.Notification,
+          cancellable: this.cancellable
         },
-        () => this.run(response)
+        (progress, token) => {
+          return this.run(response, progress, token);
+        }
```

Comment:
In terms of showing the cancel message, should we do something like this?
```typescript
// kinda pseudocode, can't remember the apis off the top of my head
(progress, token) => {
  if (this.cancellable) {
    token.onCancellation(() => wasCancelled = true)
  }
  return this.run(response, progress, token);
}

// ...

if (success) {
  notificationService.showSuccessMessage()
} else if (wasCancelled) {
  notificationService.showCancellationMessage();
} else {
  notificationService.showFailureMessage();
}
```

Reply:
Not sure the above code would make `this.run` return immediately. It depends on the underlying implementation of `this.run`.

we can move `notificationService.showSuccessMessage` and `notificationService.showFailureMessage` to the `(progress,token) => {...}` callback body, and utilize the `token.isCancellationRequested`.